CMakeLists.txt

```diff
@@ -2,7 +2,9 @@ cmake_minimum_required(VERSION 3.12)
 project(ninja)
 
 if(MSVC)
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W4 /GR- /Zc:__cplusplus")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W4 /GR- /Zc:__cplusplus -DUNICODE")
+elseif(MINGW)
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated -fdiagnostics-color -municode")
```

Comment:
Maybe like this?
```cmake
if(MSVC)
	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W4 /GR- /Zc:__cplusplus -DUNICODE")
else
	if(WIN32)
		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -municode")
	endif()
	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated -fdiagnostics-color")
endif()
```

Reply:
I agree, it looks better!

Reply:
Please do not set global cxx flags like this. If ninja is ever made into a library (one of my stated goals) this will break anything that tries to consume it.

Reply:
Also. See here for how to use generator expressions to handle compiler specific settings. https://github.com/redboltz/mqtt_cpp/blob/master/include/CMakeLists.txt

Reply:
This means that we're doing it wrong already with the current flags, right? @jonesmz As you seem to know CMake, can you create a PR which uses `add_compile_definitions` (looks to me this is the flag to use and that it only affects the current folder)?

As for the generator expressions, I would prefer to avoid them, because they look ugly AF.

Reply:
Yes, I can do that.

Generator expressions are *the* standard CMake way to do this. It's important that they be used for proper compatibility with a wide variety of internal cmake things that it does, such as proper detection for support for link-time-optimization, the creation of metadata files for a library for other cmake projects to include, and i'm sure lots of other things that I'm missing.

Reply:
Thanks.

I will read up on them then.

Reply:
https://github.com/ninja-build/ninja/pull/1679 WIP